authorization/v2/uniformResourceIdentifier.go

```diff
@@ -52,3 +110,30 @@ func getRequestArguments(ctx *gin.Context) map[string]string {
 
 	return args
 }
+
+func unmarshalArguments(source string) (map[string]string, error) {
+	arguments := strings.Split(source, "&")
+	return unmarshallURIList(arguments)
+}
+
+func unmarshallURIList(uriList []string) (map[string]string, error) {
+	toReturn := map[string]string{}
+	for _, arg := range uriList {
+		split := strings.Split(arg, "=")
+		toReturn[split[0]] = split[1]
```

Comment:
Would we have an `IndexOutOfBounds` for something like this: `test=`?

Reply:
Yep, that would happen 😨 I'll add a check here

Reply:
Nevermind, this is fine

Reply:
Are you sure? I think you are right that it would cause an error

Reply:
Ok, so it's fine for `test=` but it would break for `test`. No `=` in a key value pair would mean that the URI is malformed and we should probably return `ErrInvalidURI` here.